### DISCUSSION TITLE
Extend Server-Timing middleware to also report to OpenTelemetry

### DISCUSSION THREAD

**Discussion by thuringia** [honojs/hono#1671] [Ideas]
Hi everyone,

thanks to #1215 we can fairly easily build tracing and reporting for endpoints, but I was wondering if we can (or even should) extend it to feed other monitoring systems. Potentially there is even some overlap with the Sentry middleware here, but that's a different discussion.

How interesting would it be to have the Server Timing middleware also report calls to `setMetric` as an OpenTelemetry span?

**Comment by yusukebe** [comment]
Hi @thuringia,

Thank you for using the Server Timing Middleware.

Although I'm not very familiar with OpenTelemetry, it does sound interesting. Perhaps creating an "OpenTelemetry Middleware" as a third-party middleware would be a good solution.

**Comment by rafaell-lycan** [comment]
@yusukebe [you can check this small intro](https://newrelic.com/blog/best-practices/opentelemetry-intro-free-code-camp)
